src/sparse_blas/backends/mkl_common/mkl_operations.cxx

```diff
@@ -17,6 +17,13 @@
 * SPDX-License-Identifier: Apache-2.0
 *******************************************************************************/
 
+sycl::event optimize_gemm(sycl::queue& /*queue*/, transpose /*transpose_A*/,
+                          detail::matrix_handle* /*handle*/,
+                          const std::vector<sycl::event>& /*dependencies*/) {
+    // TODO: Call to optimize_gemm with 2024.1 oneMKL release
+    return {};
```

Comment:
We can't just return an empty sycl::event, otherwise the dependence on `dependencies` is lost. We need to return a sycl event that carries over the dependencies. Essentially:
```suggestion
    // TODO: Call to optimize_gemm with 2024.1 oneMKL release
    return queue.submit([=](sycl::handler &cgh) {
        cgh.depends_on(dependencies);
        cgh.host_task([=](){/* Empty kernel */});
    });
```

Reply:
Indeed, thanks! I've re-used your suggestion in https://github.com/oneapi-src/oneMKL/pull/403/commits/e33658c220cbcf61461c7289b770356151b09ccc

Reply:
Looks good now.